pkg/oc/admin/diagnostics/diagnostics/cluster/aggregated_logging/diagnostic.go

```diff
@@ -149,7 +148,7 @@ func (d *AggregatedLogging) Requirements() (client bool, host bool) {
 	return true, false
 }
 
-func (d *AggregatedLogging) Complete(logger *log.Logger) error {
+func (d *AggregatedLogging) selectProject() error {
```

Comment:
I think `Complete()` is still a good place for this logic. Just don't return an error if the project is not found. Then in `CanRun()`, return false and the error if `d.Project` is still `nil`. That way you get a nice skip message, not an error.

Returning an error in `Complete()` should be reserved for truly unexpected error conditions that call into question the integrity of the source code. Use the logger to inform the user without halting execution.

Reply:
Well; I see `Complete()` has been used to kick out bogus flag values too. The point is, execution stops right then if it returns an error. Be mindful that the user may be running all the diagnostics and would likely want to run the rest even if one is irrelevant.